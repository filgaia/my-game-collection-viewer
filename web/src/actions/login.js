//@vendor
import { createActions } from 'redux-actions';

const actions = createActions(
    'LOGIN_SUCCESS',
    'LOGIN_FAILURE',
    'LOGOUT_SUCCESS',
);

export {
    actions
};
