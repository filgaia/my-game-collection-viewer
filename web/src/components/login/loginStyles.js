const loginStyles = (theme) => ({
    listItem: {
        marginRight: 0
    }
});

export default loginStyles;
