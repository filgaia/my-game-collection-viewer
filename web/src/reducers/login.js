//@vendor
import Immutable from 'immutable';
import { handleActions } from 'redux-actions';
import get from 'lodash/get';
//@actions
import { actions } from '../actions/login';

const setInitialState = () => Immutable.fromJS({
    isAuthenticated: false,
    tokenId: null,
    error: ''
});

const loginReducer = handleActions(
    {
        [actions.loginSuccess]: (state, action) =>
            state.merge({
                isAuthenticated: true,
                tokenId: get(action, 'payload.tokenId', null)
            }),
        [actions.loginFailure]: (state, action) =>
            state.merge({
                isAuthenticated: false,
                tokenId: null,
                error: get(action, 'payload.error', '')
            }),
        [actions.logoutSuccess]: (state, action) =>
            state.merge({
                isAuthenticated: false,
                tokenId: null
            }),
    },
    setInitialState()
);

export default loginReducer;
