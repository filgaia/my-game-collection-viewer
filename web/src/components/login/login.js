// @vendors
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { GoogleLogin, GoogleLogout } from 'react-google-login';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
// @material
import {
  ListItemIcon,
  Tooltip,
} from '@mui/material';
import withStyles from '@mui/styles/withStyles';
// @actions
import loginActions from '../../actions/login';
// @constants
import keys from '../../data/keys.json';
// @styles
import loginStyles from './loginStyles';

export class Login extends Component {
  constructor(props) {
    super(props);

    this.buildLoginIcon = this.buildLoginIcon.bind(this);
    this.buildLogoutIcon = this.buildLogoutIcon.bind(this);
  }

  checkUserAuthenticated = () => {
    const { loginInformation } = this.props;
    loginInformation.get('isAuthenticated');
  };

  handleLoginSuccess = (response) => {
    const { loginSuccess } = this.props;
    loginSuccess({ tokenId: response.tokenId });
  };

  handleLoginFailure = (response) => {
    const { loginFailure } = this.props;
    loginFailure({ error: response.error });
  };

  buildIcon(icon, title, onClick) {
    const { classes } = this.props;

    return (
      <ListItemIcon className={classes.listItem} onClick={onClick}>
        <Tooltip title={title} aria-label={title}>
          <FontAwesomeIcon icon={icon} size="lg" />
        </Tooltip>
      </ListItemIcon>
    );
  }

  buildLoginIcon(renderProps) {
    return this.buildIcon('sign-in-alt', 'Login', renderProps.onClick);
  }

  buildLogoutIcon(renderProps) {
    return this.buildIcon('sign-out-alt', 'Logout', renderProps.onClick);
  }

  render() {
    const { logoutSuccess } = this.props;

    return this.checkUserAuthenticated() ? (
      <GoogleLogout
        clientId={keys.GOOGLE_CLIENT_ID}
        onLogoutSuccess={logoutSuccess}
        render={this.buildLogoutIcon}
      />
    )
      : (
        <GoogleLogin
          clientId={keys.GOOGLE_CLIENT_ID}
          onSuccess={this.handleLoginSuccess}
          onFailure={this.handleLoginFailure}
          render={this.buildLoginIcon}
        />
      );
  }
}

Login.propTypes = {
  classes: PropTypes.object.isRequired,
  loginFailure: PropTypes.func.isRequired,
  loginInformation: PropTypes.object.isRequired,
  loginSuccess: PropTypes.func.isRequired,
  logoutSuccess: PropTypes.func.isRequired,
};

export default connect(
  (state) => ({
    loginInformation: state.login,
  }),
  {
    loginFailure: loginActions.actions.loginFailure,
    loginSuccess: loginActions.actions.loginSuccess,
    logoutSuccess: loginActions.actions.logoutSuccess,
  },
)(withStyles(loginStyles)(Login));
