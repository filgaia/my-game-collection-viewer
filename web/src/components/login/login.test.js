import React from 'react';
import renderer from 'react-test-renderer';
import { Login } from './login';

describe('<Login />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Login
        classes={{}}
        loginFailure={jest.fn()}
        loginInformation={{ get: jest.fn() }}
        loginSuccess={jest.fn()}
        logoutSuccess={jest.fn()}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
