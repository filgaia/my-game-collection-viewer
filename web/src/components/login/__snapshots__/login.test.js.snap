// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<div
  className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
  onClick={[Function]}
/>
`;
